### ISSUE TITLE
Change the way of how destroy on deletion_protection enabled instances work

### ISSUE THREAD

**Issue by zculek-fb** [terraform-google-modules/terraform-google-sql-db#451]
### TL;DR

When destroy is applied to a Google Cloud SQL instance that has `deletion_protection` and/or `deletion_protection.enabled` set to `true`, the way it works now is that Terraform first runs a destroy on (child) resources of the Cloud SQL instance, such as user, database, etc., and when the destroy process reaches the instance itself, it fails due to `deletion_protection` being enabled.

I think the database itself is much more worth to be protected from deletion than the instance itself - the instance can easily be recreated, in a matter of minutes, but if the database is deleted and it's hundreds of GBs big, restoring from a backup could take quite some time. Since at the moment there is no way to set deletion_protection on the database itself, I suggest adding some kind of flag or an if condition or a check that would first check if deletion_protection on the instance is enabled or not, and if it is, skip deleting any of the instance resources (database, user) instead and throwing an error immediately. That would actually be much better workflow, because if `deletion_protection` is enabled on purpose, it was most probably done in an effort to prevent losing the database.

### Additional information

How it works now (deleting resources which are part of deletion_protection enabled instance):
```
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes
module.mysql_db.google_sql_user.default[0]: Destroying... [id=csql//csql]
module.mysql_db.google_sql_database.default[0]: Destroying... [id=projects/myproj-staging/instances/csql/databases/csql]
module.mysql_db.google_sql_user.default[0]: Destruction complete after 1s
module.mysql_db.random_password.user-password: Destroying... [id=none]
module.mysql_db.google_sql_database_instance.replicas["csql-replica"]: Destroying... [id=csql-read-replica]
module.mysql_db.random_password.user-password: Destruction complete after 0s
module.mysql_db.google_sql_database.default[0]: Destruction complete after 2s
╷
│ Error: Error, failed to delete instance because deletion_protection is set to true. Set it to false to proceed with instance deletion
│
│
```

Desired: (don't delete any of the resources, throw an error immediately)
```
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes
╷
│ Error: Error, failed to delete instance because deletion_protection is set to true. Set it to false to proceed with instance deletion
│
│
```

**Comment by zculek-fb** [comment]
Still pending review and comments, when someone will have time.

**Comment by zculek-fb** [comment]
> This issue is stale because it has been open 60 days with no activity. Remove stale label or comment or this will be closed in 7 days

Still pending review and comments, when someone will have time.

**Comment by imrannayer** [comment]
@zculek-fb Plz create ticket for [Google Terraform Provider](https://github.com/hashicorp/terraform-provider-google/issues). It should be handled at Provider or API level. Module cannot handle this.
